### ISSUE TITLE
Issue: nvidia-device-plugin-daemonset stuck in Init state

### ISSUE THREAD

**Issue by Heniland** [NVIDIA/gpu-operator#161]
GPU Operator vesrion = 1.5.0

### 1. Quick Debug Checklist
- [x] Are you running on an Ubuntu 18.04 node?
- [x] Are you running Kubernetes v1.13+?
- [x] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- [ ] Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?

### 1. Issue or feature description
nvidia-device-plugin-daemonset-9w58x stuck and in my gpu pod (shlomo-predict) stays "Pending"  

rebooting the node doesn't help 

#### The workaround I have for now is:

- scale down my pods that use GPU
- uninstall GPU operator using helm
- installing it again using helm
- scale up the pods that use GPU

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

- [x] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)

```
 kubectl describe clusterpolicies --all-namespaces
Name:         cluster-policy
Namespace:
Labels:       app.kubernetes.io/component=gpu-operator
              app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: gpu-operator
              meta.helm.sh/release-namespace: default
API Version:  nvidia.com/v1
Kind:         ClusterPolicy
Metadata:
  Creation Timestamp:  2021-03-04T13:51:03Z
  Generation:          1
  Managed Fields:
    API Version:  nvidia.com/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:meta.helm.sh/release-name:
          f:meta.helm.sh/release-namespace:
        f:labels:
          .:
          f:app.kubernetes.io/component:
          f:app.kubernetes.io/managed-by:
      f:spec:
        .:
        f:dcgmExporter:
          .:
          f:args:
          f:image:
          f:imagePullPolicy:
          f:nodeSelector:
            .:
            f:nvidia.com/gpu.present:
          f:repository:
          f:version:
        f:devicePlugin:
          .:
          f:args:
          f:image:
          f:imagePullPolicy:
          f:nodeSelector:
            .:
            f:nvidia.com/gpu.present:
          f:repository:
          f:securityContext:
            .:
            f:privileged:
          f:version:
        f:driver:
          .:
          f:image:
          f:imagePullPolicy:
          f:nodeSelector:
            .:
            f:nvidia.com/gpu.present:
          f:repository:
          f:securityContext:
            .:
            f:privileged:
            f:seLinuxOptions:
              .:
              f:level:
          f:tolerations:
          f:version:
        f:gfd:
          .:
          f:discoveryIntervalSeconds:
          f:image:
          f:imagePullPolicy:
          f:migStrategy:
          f:nodeSelector:
            .:
            f:nvidia.com/gpu.present:
          f:repository:
          f:version:
        f:operator:
          .:
          f:defaultGPUMode:
          f:defaultRuntime:
          f:validator:
            .:
            f:image:
            f:imagePullPolicy:
            f:repository:
            f:version:
        f:toolkit:
          .:
          f:image:
          f:imagePullPolicy:
          f:nodeSelector:
            .:
            f:nvidia.com/gpu.present:
          f:repository:
          f:securityContext:
            .:
            f:privileged:
            f:seLinuxOptions:
              .:
              f:level:
          f:tolerations:
          f:version:
    Manager:      Go-http-client
    Operation:    Update
    Time:         2021-03-04T13:51:03Z
    API Version:  nvidia.com/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:state:
    Manager:         gpu-operator
    Operation:       Update
    Time:            2021-03-04T13:51:08Z
  Resource Version:  147492
  UID:               5d5bb07a-69b5-4386-b900-0dc9908db1ea
Spec:
  Dcgm Exporter:
    Args:
      -f
      /etc/dcgm-exporter/dcp-metrics-included.csv
    Image:              dcgm-exporter
    Image Pull Policy:  IfNotPresent
    Node Selector:
      nvidia.com/gpu.present:  true
    Repository:                nvcr.io/nvidia/k8s
    Version:                   2.0.13-2.1.2-ubuntu20.04
  Device Plugin:
    Args:
      --mig-strategy=none
      --pass-device-specs=false
      --fail-on-init-error=true
      --device-list-strategy=envvar
    Image:              k8s-device-plugin
    Image Pull Policy:  IfNotPresent
    Node Selector:
      nvidia.com/gpu.present:  true
    Repository:                nvcr.io/nvidia
    Security Context:
      Privileged:  true
    Version:       v0.7.3
  Driver:
    Image:              driver
    Image Pull Policy:  IfNotPresent
    Node Selector:
      nvidia.com/gpu.present:  true
    Repository:                nvcr.io/nvidia
    Security Context:
      Privileged:  true
      Se Linux Options:
        Level:  s0
    Tolerations:
      Effect:    NoSchedule
      Key:       nvidia.com/gpu
      Operator:  Exists
    Version:     450.80.02
  Gfd:
    Discovery Interval Seconds:  60
    Image:                       gpu-feature-discovery
    Image Pull Policy:           IfNotPresent
    Mig Strategy:                none
    Node Selector:
      nvidia.com/gpu.present:  true
    Repository:                nvcr.io/nvidia
    Version:                   v0.3.0
  Operator:
    Default GPU Mode:  passthrough
    Default Runtime:   docker
    Deploy GFD:        true
    Validator:
      Image:              cuda-sample
      Image Pull Policy:  IfNotPresent
      Repository:         nvcr.io/nvidia/k8s
      Version:            vectoradd-cuda10.2
  Toolkit:
    Image:              container-toolkit
    Image Pull Policy:  IfNotPresent
    Node Selector:
      nvidia.com/gpu.present:  true
    Repository:                nvcr.io/nvidia/k8s
    Security Context:
      Privileged:  true
      Se Linux Options:
        Level:  s0
    Tolerations:
      Key:       CriticalAddonsOnly
      Operator:  Exists
      Effect:    NoSchedule
      Key:       nvidia.com/gpu
      Operator:  Exists
    Version:     1.4.2-ubuntu18.04
Status:
  State:  ready
Events:   <none>
```

## The issue:
```
kubectl get pods -A
NAMESPACE                NAME                                                          READY   STATUS      RESTARTS   AGE
kube-system              helm-install-traefik-4w6gp                                    0/1     Completed   0          44h
gpu-operator-resources   nvidia-device-plugin-validation                               0/1     Completed   0          44h
prometheus               prometheus-kube-prometheus-operator-8cfb4bbd6-xhhjd           1/1     Running     1          44h
default                  gpu-operator-node-feature-discovery-master-7994f664cc-pxqxs   1/1     Running     1          44h
prometheus               svclb-prometheus-grafana-kg2w4                                1/1     Running     1          44h
kube-system              svclb-traefik-ksj2x                                           2/2     Running     2          44h
prometheus               prometheus-prometheus-kube-prometheus-prometheus-0            2/2     Running     3          44h
kube-system              coredns-854c77959c-8nrj2                                      1/1     Running     1          44h
kube-system              local-path-provisioner-5ff76fc89d-xk9bw                       1/1     Running     1          44h
default                  gpu-operator-c7555d786-jjhbp                                  1/1     Running     2          44h
default                  influxdb-0                                                    1/1     Running     1          44h
prometheus               prometheus-grafana-8ff59c97-s89lw                             2/2     Running     2          44h
kube-system              traefik-6f9cbd9bd4-6z9gr                                      1/1     Running     1          44h
kube-system              metrics-server-86cbb8457f-fdv2c                               1/1     Running     2          44h
default                  gpu-operator-node-feature-discovery-worker-tbkf5              1/1     Running     2          44h
default                  gpu-operator-node-feature-discovery-worker-8595p              1/1     Running     2          44h
prometheus               svclb-prometheus-grafana-24ql2                                1/1     Running     2          44h
default                  shlomo-predict-857868ff-xvmtx                                 0/2     Pending     0          21m
gpu-operator-resources   nvidia-device-plugin-daemonset-hhwv7                          0/1     Init:0/1    2          44h
gpu-operator-resources   nvidia-driver-daemonset-wl2n2                                 1/1     Running     2          44h
gpu-operator-resources   nvidia-dcgm-exporter-r68gv                                    0/1     Init:0/1    2          44h
gpu-operator-resources   gpu-feature-discovery-zhlbs                                   1/1     Running     2          44h
kube-system              svclb-traefik-6hsrv                                           2/2     Running     4          44h
gpu-operator-resources   nvidia-container-toolkit-daemonset-8dk69                      1/1     Running     2          44h
```
## After reinstall: 
 - [x] kubernetes pods status: `kubectl get pods --all-namespaces`
 

```
kubectl get pods --all-namespaces
NAMESPACE                NAME                                                          READY   STATUS      RESTARTS   AGE
kube-system              helm-install-traefik-4w6gp                                    0/1     Completed   0          45h
kube-system              svclb-traefik-ksj2x                                           2/2     Running     2          45h
kube-system              coredns-854c77959c-8nrj2                                      1/1     Running     1          45h
kube-system              local-path-provisioner-5ff76fc89d-xk9bw                       1/1     Running     1          45h
default                  influxdb-0                                                    1/1     Running     1          45h
kube-system              traefik-6f9cbd9bd4-6z9gr                                      1/1     Running     1          45h
kube-system              metrics-server-86cbb8457f-fdv2c                               1/1     Running     2          45h
kube-system              svclb-traefik-6hsrv                                           2/2     Running     6          45h
prometheus               svclb-prometheus-grafana-l7m6q                                1/1     Running     0          36m
prometheus               svclb-prometheus-grafana-bgkt6                                1/1     Running     0          36m
prometheus               prometheus-kube-prometheus-operator-8cfb4bbd6-vl655           1/1     Running     0          36m
prometheus               prometheus-prometheus-kube-prometheus-prometheus-0            2/2     Running     1          36m
prometheus               prometheus-grafana-8ff59c97-ph27f                             2/2     Running     0          36m
default                  gpu-operator-node-feature-discovery-worker-nph5k              1/1     Running     0          21m
default                  gpu-operator-node-feature-discovery-master-7994f664cc-jgk7n   1/1     Running     0          21m
default                  gpu-operator-node-feature-discovery-worker-b2sbx              1/1     Running     0          21m
gpu-operator-resources   gpu-feature-discovery-w2b9h                                   1/1     Running     0          21m
default                  gpu-operator-c7555d786-tj5pb                                  1/1     Running     0          21m
gpu-operator-resources   nvidia-driver-daemonset-9v45k                                 1/1     Running     0          21m
gpu-operator-resources   nvidia-container-toolkit-daemonset-t72q9                      1/1     Running     0          21m
gpu-operator-resources   nvidia-device-plugin-daemonset-9w58x                          1/1     Running     0          21m
gpu-operator-resources   nvidia-device-plugin-validation                               0/1     Completed   0          21m
gpu-operator-resources   nvidia-dcgm-exporter-lmk7t                                    1/1     Running     0          20m
```
 - [x] kubernetes daemonset status: `kubectl get ds --all-namespaces`

```
 kubectl get ds --all-namespaces
NAMESPACE                NAME                                         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                 AGE
kube-system              svclb-traefik                                2         2         2       2            2           <none>                        45h
prometheus               svclb-prometheus-grafana                     2         2         2       2            2           <none>                        37m
default                  gpu-operator-node-feature-discovery-worker   2         2         2       2            2           <none>                        22m
gpu-operator-resources   gpu-feature-discovery                        1         1         1       1            1           nvidia.com/gpu.present=true   22m
gpu-operator-resources   nvidia-driver-daemonset                      1         1         1       1            1           nvidia.com/gpu.present=true   22m
gpu-operator-resources   nvidia-container-toolkit-daemonset           1         1         1       1            1           nvidia.com/gpu.present=true   22m
gpu-operator-resources   nvidia-device-plugin-daemonset               1         1         1       1            1           nvidia.com/gpu.present=true   21m
gpu-operator-resources   nvidia-dcgm-exporter                         1         1         1       1            1           nvidia.com/gpu.present=true   21m
```
 - [x] Docker configuration file: `cat /etc/docker/daemon.json`
 
```
cat /etc/docker/daemon.json
{
    "default-runtime": "nvidia",
    "runtimes": {
        "nvidia": {
            "args": [],
            "path": "/usr/local/nvidia/toolkit/nvidia-container-runtime"
        }
    }
}
```
 - [x] Docker runtime configuration: `docker info | grep runtime`

```
sudo docker info | grep Runtime
Runtimes: runc nvidia
Default Runtime: nvidia
WARNING: No swap limit support
```

**Comment by shivamerla** [comment]
@Heniland This was resolved in 1.5.2. Please try with that or latest 1.6.0.

**Comment by Heniland** [comment]
I checked and confirmed that the update to 1.6.0 does solve the issue, it's working for me. Thank you @shivamerla
